src/widget/vumeterpool.cpp

```diff
@@ -0,0 +1,43 @@
+#include "widget/vumeterpool.h"
+
+#include <cassert>
+
+#include "widget/wvumeter.h"
+
+VuMeterPool::VuMeterPool() {
+}
+
+VuMeterPool::~VuMeterPool() {
+    destroyAdopted();
+    assert(m_pVuMeters.isEmpty());
+}
+
+void VuMeterPool::destroyAdopted() {
+    auto it = m_pVuMeters.begin();
+    while (it != m_pVuMeters.end()) {
+        auto pVuMeter = *it;
+        if (pVuMeter->parent()) {
+            it++;
+        } else {
+            it = m_pVuMeters.erase(it);
+            delete pVuMeter;
+        }
+    }
+}
+
+void VuMeterPool::add(WVuMeter* pVuMeter) {
+    m_pVuMeters.append(pVuMeter);
+}
+
+void VuMeterPool::remove(WVuMeter* pVuMeter) {
+    m_pVuMeters.removeOne(pVuMeter);
+}
+
+void VuMeterPool::adopt() {
+    // Remove the parent of the vumeters,, but we keep their pointers,
+    // effectively "adopting" them.
+    for (auto pVuMeter : m_pVuMeters) {
+        pVuMeter->disconnect();
+        pVuMeter->setParent(Q_NULLPTR);
```

Comment:
```suggestion
        pVuMeter->setParent(nullptr);
```
The macro has been introduced for the transition to C++11 we are now at C++17

Reply:
I know. I reckoned that in qt specific code it was preferable, style-wise, to use the qt specific macros, but I am happy to use nullptr. Anyway, all to be reverted.